service/watcher/configmap.go

```diff
@@ -99,25 +100,34 @@ func (c *AppValueWatcher) watch(ctx context.Context) {
 				}
 			}
 
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("listed apps depends on %#q configmap in namespace %#q", cm.Name, cm.Namespace))
 			for app := range storedIndex {
 				c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("triggering %#q app update in namespace %#q", app.Name, app.Namespace))
 
-				err := c.addAnnotation(ctx, app, cm.GetResourceVersion())
+				err := c.addAnnotation(ctx, app, cm.GetResourceVersion(), configMapType)
 				if err != nil {
 					c.logger.LogCtx(ctx, "level", "info", "message", fmt.Sprintf("failed to add annotation to app %#q in namespace %#q", app.Name, app.Namespace), "stack", fmt.Sprintf("%#v", err))
 					continue
 				}
 
 				c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("triggered %#q app update in namespace %#q", app.Name, app.Namespace))
 			}
+			c.logger.LogCtx(ctx, "level", "debug", "message", fmt.Sprintf("listed apps depends on %#q configmap in namespace %#q", cm.Name, cm.Namespace))
 		}
 
 		c.logger.Log("debug", "watch channel had been closed, reopening...")
 	}
 }
 
-func (c *AppValueWatcher) addAnnotation(ctx context.Context, app appIndex, latestResourceVersion string) error {
-	versionAnnotation := fmt.Sprintf("%s/%s", annotation.AppOperatorPrefix, annotation.LatestConfigMapVersion)
+func (c *AppValueWatcher) addAnnotation(ctx context.Context, app appIndex, latestResourceVersion string, resType resourceType) error {
+	var versionAnnotation string
+	{
+		if resType == configMapType {
```

Comment:
To support `configmap`, `secret` at the same time, adding an `if` statement like this would work for most cases.